dir-overlay/module-artifact-gen/dir-overlay-artifact-gen

```diff
@@ -0,0 +1,170 @@
+#!/bin/bash
+
+set -e
+
+show_help() {
+  cat << EOF
+
+Simple tool to generate Mender Artifact suitable for dir-overlay Update Module
+
+Usage: $0 [options] file-tree
+
+    Options: [ -n|artifact-name -t|--device-type -d|--dest-dir -o|--output_path -h|--help ]
+
+        --artifact-name     - Artifact name
+        --device-type       - Target device type identification
+        --dest-dir          - Target destination directory where to deploy the update (Defaults to "/")
+        --output-path       - Path to output file. Default: file-install-artifact.mender
+        --help              - Show help and exit
+        file-tree           - Directory tree to bundle in the update
+
+Common use-case for the dir-overlay is to install files overlayed
+on the root file-system.
+
+Example tree to deploy:
+
+    $ tree
+    .
+    ├── etc
+    │   └── app.conf
+    └── usr
+        ├── bin
+        │   └── app
+        └── share
+            └── app
+                └── data.txt
+
+EOF
+}
+
+show_help_and_exit_error() {
+  show_help
+  exit 1
+}
+
+check_dependency() {
+  hash "$1" || \
+        (echo "Fatal error: $1 not found in PATH." && exit 1)
+}
+
+check_dependency mender-artifact
+
+device_type=""
+artifact_name=""
+dest_dir="/"
+dest_dir_file="dest_dir"
+output_path="file-install-artifact.mender"
+update_files_tar="update.tar"
+directories_file="directories"
+manifest_file="manifest"
+file_tree=""
+
+while (( "$#" )); do
+  case "$1" in
+    --device-type | -t)
+      if [ -z "$2" ]; then
+        show_help_and_exit_error
+      fi
+      device_type=$2
+      shift 2
+      ;;
+    --artifact-name | -n)
+      if [ -z "$2" ]; then
+        show_help_and_exit_error
+      fi
+      artifact_name=$2
+      shift 2
+      ;;
+    --dest-dir | -d)
+      if [ -z "$2" ]; then
+        show_help_and_exit_error
+      fi
+      dest_dir=$2
+      shift 2
+      ;;
+    --output-path | -o)
+      if [ -z "$2" ]; then
+        show_help_and_exit_error
+      fi
+      output_path=$2
+      shift 2
+      ;;
+    -h | --help)
+      show_help
+      exit 0
+      ;;
+    -*)
+      echo "Error: unsupported option $1"
+      show_help_and_exit_error
+      ;;
+    *)
+      if [ -n "$file_tree" ]; then
+        echo "File tree already specified. Unrecognized argument \"$1\""
+        show_help_and_exit_error
+      fi
+      file_tree="$1"
+      shift
+      ;;
+  esac
+done
+
+if [ -z "${artifact_name}" ]; then
+  echo "Artifact name not specified. Aborting."
+  show_help_and_exit_error
+fi
+
+if [ -z "${device_type}" ]; then
+  echo "Device type not specified. Aborting."
+  show_help_and_exit_error
+fi
+
+if [ -z "${file_tree}" ]; then
+  echo "File tree not specified. Aborting."
+  show_help_and_exit_error
+fi
+
+# Check dest-dir is an absolute path
+case $dest_dir in
+  /*)
+    ;;
+  *)
+    echo "Destination dir must be an absolute path. Aborting"
+    exit 1
+  ;;
+esac
+
+# Create tarball, accepts single file or directory.
```

Comment:
This comment (and part of the logic below) is now obsolete. You can simplify it by just testing that it is a directory

Reply:
Thanks, fixed